src/comp/comp.py

```diff
@@ -25,47 +25,70 @@ def __repr__(self):
 # whose name starts with 'D':
 print("Starts with D:")
 a = []
+
+for d in humans:
+    if d.name[0] == "D":
+        a.append(d.name)
 print(a)
 
 # Write a list comprehension that creates a list of names of everyone
 # whose name ends in "e".
 print("Ends with e:")
 b = []
+for e in humans:
+    if e.name[-1] == "e":
+        b.append(e.name)
 print(b)
 
 # Write a list comprehension that creates a list of names of everyone
 # whose name starts with any letter between 'C' and 'G' inclusive.
 print("Starts between C and G, inclusive:")
 c = []
+for cg in humans:
+    if cg.name[0].lower() in ["c", "d", "e", "f", "g"]:
+        c.append(cg.name)
 print(c)
 
 # Write a list comprehension that creates a list of all the ages plus 10.
 print("Ages plus 10:")
 d = []
+for ageTen in humans:
+    d.append(ageTen.age+10)
 print(d)
 
 # Write a list comprehension that creates a list of strings which are the name
 # joined to the age with a hyphen, for example "David-31", for all humans.
 print("Name hyphen age:")
 e = []
+for hyphen in humans:
+    e.append(f"{hyphen.name}-{hyphen.age}")
 print(e)
 
 # Write a list comprehension that creates a list of tuples containing name and
 # age, for example ("David", 31), for everyone between the ages of 27 and 32,
 # inclusive.
 print("Names and ages between 27 and 32:")
 f = []
+for tuples in humans:
+    if tuples.age > 26 and tuples.age < 33:
+        insideTuple = (tuples.name, tuples.age)
+        f.append(insideTuple)
 print(f)
 
 # Write a list comprehension that creates a list of new Humans like the old
 # list, except with all the names uppercase and the ages with 5 added to them.
 # The "humans" list should be unmodified.
 print("All names uppercase:")
 g = []
+for newHuman in humans:
+    insideHuman = Human(newHuman.name.upper(), newHuman.age+5)
+    g.append(insideHuman)
 print(g)
 
 # Write a list comprehension that contains the square root of all the ages.
 print("Square root of ages:")
 import math
 h = []
+for newMath in humans:
+    h.append(math.sqrt(newMath.age))
 print(h)
```

Comment:
My dude, um, none of these are list comprehensions. True, they my all pass the tests (lol, and again TDD is BS) but they do not actually answer the questions. Please revise.